docs/functional/architecture/0011-Scalar_vs_0d_Fields.md

```diff
@@ -0,0 +1,122 @@
+
+```python=
+@field_operator
+def foo() -> Field[[], float]
+    return 1.0
+```
+
+An explicit broadcast would be required to make this correct.
+
+```python=
+@field_operator
+def foo() -> Field[[], float]
+    return broadcast(1.0, ())
+```
+
+#### Implicit broadcast to Field
+
+In the following call an implicit broadcast to Field is performed in the decorator (the annotation influences behavior).
+
+```python=
+@field_operator
+def foo(scalar: Field[[], float]):
+    return scalar
+
+foo(1.0)
+```
+
+Note that the following is also valid as no broadcasting to `Field` is applied.
+
+```python=
+@field_operator
+def foo(scalar: float) -> float:
+    return scalar
+
+foo(1.0)
+```
+
+This enables propagating scalars for cases where a scalars are required.
+
+#### 0-d fields cannot be used where a scalar is expected
+
+The following is forbidden because ternaries only work with scalar conditions.
+
+```python=
+@field_operator
+def foo(cond: Field[[], int]):
+    return some_field_op() if cond == 1 else some_other_field_op()
+```
+
+If needed, we might add functionality to extract the scalar value from a 0-d field.
+
+### Notes on lowering
+
+This decision doesn't change the current definition of iterator ir or the `fn` C++ backend:
+- a stencil (any lowered field_operator) will only accept iterators (i.e. `deref` is required);
+- `fn` will accept SIDs, scalars have to we wrapped in a helper, e.g. `global_parameter`.
+
+## Alternatives Considered
+
+### Scalar `T` and 0-d field `Field[[], T]` are aliases for each other
+
+Making scalars and 0-d field the same (aliases for each other) simplifies usage of the language in many places.
+
+Example: A reduction over a dimension of a field returns a field with one dimension less. Reducing over a 1-d field will therefore return a 0-d field, but that value might be needed in a place where a scalar is expected.
+
+However, we would need implement it with either possibly strange side-effects (making builtin types, e.g. `float` a subclass of Field) or create custom types for scalars (which is not desirable from a usage perspective, e.g. `gt4py.float`) to make this properly work with Python typing.
+
+It would allow to make some cases work that are forbidden by the now decided rules, e.g. 
+
+```python=
+@field_operator
+def foo() -> Field[[], float]
+    return 1.0
+```
+
+would be correct, e.g. if we implement `Field[[], float]` as a protocol that matches `1.0`.
+
+If we implement our own types, we have full freedom, to really make them aliases and get the following semantic work
+
+```python=
+scalar: gt4py.float(4.0)
+assert isinstance(scalar, gt4py.float)
+assert isinstance(scalar, Field[[], gt4py.float])
+assert scalar.dtype == gt4py.float
+assert scalar.dtype == Field[[], gt4py.float]
+assert scalar.dtype.dtype.dtype.dtype == gt4py.float
+assert scalar.dtype.dtype.dtype.dtype == Field[[], gt4py.float]
+```
+
```

Comment:
I would add another alternative here (as an ideal solution for the future): fields are different than scalars, but 0-d fields implement the appropriate scalar conversion operator matching its data type  (e.g. `__bool__`, `__int__`, `__float__`). In this way, 0-d field would also behave as scalar values in Python code. This is the approach used in the Python array API standard (https://data-apis.org/array-api/latest/API_specification/array_object.html). Additionally, although I personally think we should define our own scalar types, this alternative doesn't require it.

```python

a_0d_field

assert isinstance(a_0d_field, Field)

if a_0d_field > 3.5:
    ...
else:
    ...

```

Reply:
Can you do the edit? Probably you can formulate that better...

Reply:
Done